packages/core/src/browser/label-provider.ts

```diff
@@ -86,12 +92,31 @@ export class DefaultUriLabelProviderContribution implements LabelProviderContrib
 }
 
 @injectable()
-export class LabelProvider {
+export class LabelProvider implements Disposable {
```

Comment:
`implements Disposable` is it required? `LabelProvider` lifecycle is bound to the app, it cannot be disposed.